### PR TITLE
Improves Stacktrace class and added serializer test cases

### PR DESCRIPTION
**Pull request by khaydarov**

### CODE REVIEW

File: src/Addons/AddonInterface.php [hunk 1]
```diff
@@ -11,6 +11,13 @@
  */
 interface AddonInterface
 {
+    /**
+     * Returns addon group name
```
Comment (neSpecc) [inline src/Addons/AddonInterface.php:15]:
```suggestion
     * Returns addon name
```

File: src/Catcher.php [hunk 2]
```diff
@@ -149,10 +140,16 @@ public function sendEvent(array $payload): void
     private function __construct(array $options)
     {
         $options = new Options($options);
-        $factory = new EventPayloadFactory();
+
+        $serializer = new Serializer();
```
Comment (neSpecc) [inline src/Catcher.php:144]: add comments please about this section

File: src/Catcher.php [hunk 3]
```diff
@@ -149,10 +140,16 @@ public function sendEvent(array $payload): void
...
+        $stacktraceBuilder = new StacktraceFrameBuilder($serializer);
+
+        $builder = new EventPayloadBuilder($stacktraceBuilder);
```
Comment (neSpecc) [inline src/Catcher.php:147]: and this

File: src/Event.php [hunk 4]
```diff
@@ -5,6 +5,8 @@
 namespace Hawk;
 
 /**
+ * Event value-object
```
Comment (neSpecc) [inline src/Event.php:8]: Please provide more understandable description

File: src/EventPayloadBuilder.php [hunk 5]
```diff
@@ -22,12 +20,29 @@ class EventPayloadFactory
      */
     private $addonsResolvers = [];
 
+    /**
+     * @var StacktraceFrameBuilder
```
Comment (neSpecc) [inline src/EventPayloadBuilder.php:24]: description missed

File: src/EventPayloadBuilder.php [hunk 6]
```diff
@@ -22,12 +20,29 @@ class EventPayloadFactory
...
     /**
      * EventPayloadFactory constructor.
      */
-    public function __construct()
+    public function __construct(StacktraceFrameBuilder $stacktraceFrameBuilder)
+    {
+        $this->stacktraceFrameBuilder = $stacktraceFrameBuilder;
+    }
+
+    /**
```
Comment (neSpecc) [inline src/EventPayloadBuilder.php:36]: description missed

File: src/EventPayloadBuilder.php [hunk 7]
```diff
@@ -22,12 +20,29 @@ class EventPayloadFactory
...
+     * @param AddonInterface $addon
```
Comment (neSpecc) [inline src/EventPayloadBuilder.php:37]: description missed

File: src/Serializer.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hawk;
+
+/**
+ * Class Serializer
```
Comment (neSpecc) [inline src/Serializer.php:8]: description repeats the name. Please write _what this class is doing_

File: tests/Unit/StacktraceFrameBuilderTest.php [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+class StacktraceFrameBuilderTest extends TestCase
+{
+    public function testResultingStacktraceFrames(): void
+    {
+        $serializer = new Serializer();
+        $fixture = new StacktraceFrameBuilder($serializer);
+        $exception = new \Exception();
+
+        $stacktrace = $fixture->buildStack($exception);
+        $this->assertCount(11, $stacktrace);
```
Comment (neSpecc) [inline tests/Unit/StacktraceFrameBuilderTest.php:20]: where `11` came from?

File: tests/Unit/SerializerTest.php [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+namespace Hawk\Tests\Unit;
+
+use Hawk\Serializer;
+use PHPUnit\Framework\TestCase;
+
+class SerializerTest extends TestCase
+{
+    /**
+     * @dataProvider valueProvider
```
Comment (neSpecc) [inline tests/Unit/SerializerTest.php:13]: what does it mean? Pls add a comment

Comment (khaydarov) [reply tests/Unit/SerializerTest.php:13]: it is standard notation for phpUnit. Data provider passes consequently values from array and executes test

**Review by neSpecc — APPROVED** [review]

**Review by talyguryn — APPROVED** [review]

**Pull request merged**